### PR TITLE
feat: Volcano Supports K8s v1.31

### PR DESCRIPTION
**Pull request by vie-serendipity**
I split the content into four commits.
1. change go mod
2. upgrade KUBE_VERSION
3. solve some api changes after adaptation
4. sync changes regarding volumebing

I have two tasks to be completed.
- [x] introduce the updated version of volcano apis, because the pr hasn't been merged yet
- [x] update README after running e2e tests, unit tests has passed already

### CODE REVIEW

**Comment by Monokaix** [comment]
great job!

**Comment by Monokaix** [comment]
You can also refer to 1.30 adaptation: )

**Comment by vie-serendipity** [comment]
I'll fix these issues to pass the test.

**Comment by Monokaix** [comment]
You can also refer to 1.30 adaptation https://github.com/volcano-sh/volcano/pull/3556 : )

File: pkg/scheduler/capabilities/volumebinding/assume_cache_test.go [hunk 1]
```diff
@@ -24,9 +24,10 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/component-helpers/storage/volume"
 	"k8s.io/klog/v2/ktesting"
+	"k8s.io/kubernetes/pkg/scheduler/util/assumecache"
```
Comment (Monokaix) [inline pkg/scheduler/capabilities/volumebinding/assume_cache_test.go:27]: Thanks for your contribution！are volumebinding codes synced from k8s?

**Comment by Monokaix** [comment]
Please notice that to keep API compatibility，we should modify the code `CSIStorageCapacityInformer: fh.SharedInformerFactory().Storage().V1().CSIStorageCapacities()` in file pkg/scheduler/capabilities/volumebinding/volume_binding.go, and change from V1() to V1beta1() and test whether v1beat1 is supported in k8s v1.31: )

File: pkg/scheduler/capabilities/volumebinding/assume_cache_test.go (hunk 1 above)
Comment (vie-serendipity) [reply pkg/scheduler/capabilities/volumebinding/assume_cache_test.go:27]: done

**Comment by vie-serendipity** [comment]
@Monokaix  Thanks for your comment, I've already updated the volume binding-related code and changed v1 to v1beta1.

**Comment by Monokaix** [comment]
Please update the task lists in description: )

**Comment by Monokaix** [comment]
cc @wangyang0616 @lowang-bh @hwdef

File: .github/workflows/e2e_vcctl.yaml [hunk 2]
```diff
@@ -31,8 +31,8 @@ jobs:
 
       - name: Install dependences
         run: |
-          GO111MODULE="on" go install sigs.k8s.io/kind@v0.23.0
-          curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.30.1/bin/linux/amd64/kubectl && sudo install kubectl /usr/local/bin/kubectl
+          GO111MODULE="on" go install sigs.k8s.io/kind@v0.24.0
+          curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.31.0/bin/linux/amd64/kubectl && sudo install kubectl /usr/local/bin/kubectl
```
[the remark this answers is not in the document]
Comment (Monokaix) [reply .github/workflows/e2e_vcctl.yaml:35]: seems no downgrade: )

[the remark this answers is not in the document]
Comment (hwdef) [reply .github/workflows/e2e_vcctl.yaml:35]: Sorry I misread it.

**Comment by hwdef** [comment]
please squash your commit

File: pkg/scheduler/capabilities/volumebinding/binder.go [hunk 3]
```diff
@@ -1010,12 +1013,6 @@ func (b *volumeBinder) revertAssumedPVCs(claims []*v1.PersistentVolumeClaim) {
 // hasEnoughCapacity checks whether the provisioner has enough capacity left for a new volume of the given size
 // that is available from the node.
 func (b *volumeBinder) hasEnoughCapacity(logger klog.Logger, provisioner string, claim *v1.PersistentVolumeClaim, storageClass *storagev1.StorageClass, node *v1.Node) (bool, error) {
-	// This is an optional feature. If disabled, we assume that
-	// there is enough storage.
-	if !b.capacityCheckEnabled {
```
Comment (ouyangshengjia) [inline pkg/scheduler/capabilities/volumebinding/binder.go:1015]: Was this code deleted by mistake?
**b.csiDriverLister** and **b.csiStorageCapacityLister** is null when **capacityCheckEnabled == false**

Comment (vie-serendipity) [reply pkg/scheduler/capabilities/volumebinding/binder.go:1015]: yes, it's fixed now.

**Pull request merged**